ArduPlane/quadplane.cpp

```diff
@@ -4114,23 +4114,16 @@ float QuadPlane::get_land_airspeed(void)
     if (qstate == QPOS_APPROACH ||
         plane.control_mode == &plane.mode_rtl) {
         const float cruise_speed = plane.aparm.airspeed_cruise_cm * 0.01;
-        float approach_speed = cruise_speed;
-        float tecs_land_airspeed = plane.TECS_controller.get_land_airspeed();
-        if (is_positive(tecs_land_airspeed)) {
-            approach_speed = tecs_land_airspeed;
-        } else {
-            if (qstate == QPOS_APPROACH) {
-                // default to half way between min airspeed and cruise
-                // airspeed when on the approach
-                approach_speed = 0.5*(cruise_speed+plane.aparm.airspeed_min);
-            } else {
-                // otherwise cruise
+        float approach_speed = plane.TECS_controller.get_land_airspeed();    
+        if (qstate != QPOS_APPROACH) {
```

Comment:
This logic is not quite the same. Previously we would always use the `LAND_ARSPD` parameter if set. Now we only use it if not in approach. 

Unfortunately I don't see a good way to fix it.

Maybe the change in behavior is OK in any-case.

Reply:
it still uses it if its set...look at the get_land_airspeed function

Reply:
Previously `tecs_land_airspeed` (`TECS_LAND_ARSPD`) was always used if positive. Now we only use it if positive AND not in approach.

So now in approach we return at `cruise_speed` and not `tecs_land_airspeed`.

There is a change in behaviour. 

As I say, I'm not sure if its good or bad.

Reply:
sorry, it is a minor additional change (back to original RTL speed behaviour before the change to interpolated landing airspeed changed it for all of the RTL)...but a good one...